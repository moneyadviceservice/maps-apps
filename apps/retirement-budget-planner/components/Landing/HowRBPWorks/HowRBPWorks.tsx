import { Heading } from '@maps-react/common/components/Heading';
import { Link } from '@maps-react/common/components/Link';
import { ListElement } from '@maps-react/common/components/ListElement';
import { Paragraph } from '@maps-react/common/components/Paragraph';

export const HowRBPWorks = () => (
  <section data-testid="how-rbp-works">
    <Heading level="h2" variant="secondary" className="text-[38px] mb-8 mt-2">
      How our Retirement budget planner works
    </Heading>
    <Paragraph>
      Our Retirement budget planner compares your income to your outgoings and
      calculates if you’ll have any money left over.
    </Paragraph>
    <Paragraph>
      You’ll also see a breakdown of where you spend your money to help identify
      if you can cut costs.
    </Paragraph>
    <Paragraph>The information you’ll need</Paragraph>
    <Paragraph className="font-bold"> You’ll be asked to enter:</Paragraph>
    <ListElement
      items={[
        'the age you would like to retire.',
        'your current income.',
        'how much you spend, including household bills and paying into your pension.',
        'an email address if you wish to save and come back later. ',
      ]}
      color="blue"
      variant="unordered"
      className="pb-8 pl-2 text-sm list-inside "
    />
    <Paragraph>
      You can create a household budget by adding these details for another
      person.
    </Paragraph>
    <Link
      data-testid="rbp-link-from-how-it-works"
      className="my-4"
      asButtonVariant="primary"
      id="submit"
      href="/about-you"
    >
      Start Retirement Budget Planner
    </Link>
    <Paragraph className="pt-4 pb-8">
      It should take about five minutes to complete.
    </Paragraph>
  </section>
);
