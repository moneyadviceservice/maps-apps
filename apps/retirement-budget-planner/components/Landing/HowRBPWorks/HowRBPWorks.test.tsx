import { render, screen, within } from '@testing-library/react';

import { HowRBPWorks } from '../HowRBPWorks/HowRBPWorks';

import '@testing-library/jest-dom';

describe('test HowRBPWorks component', () => {
  it('should render the component', () => {
    render(<HowRBPWorks />);
    expect(screen).toMatchSnapshot();
    expect(
      screen.getByText('How our Retirement budget planner works'),
    ).toBeInTheDocument();
    expect(
      screen.getByText(
        'You can create a household budget by adding these details for another person.',
      ),
    ).toBeInTheDocument();
  });
  it('should render the list elements correctly', () => {
    render(<HowRBPWorks />);
    const list = screen.getByTestId('list-element');

    const listItems = within(list).getAllByRole('listitem');
    expect(listItems[0]).toHaveTextContent('the age you would like to retire.');
    expect(list).toMatchSnapshot();
  });
});
