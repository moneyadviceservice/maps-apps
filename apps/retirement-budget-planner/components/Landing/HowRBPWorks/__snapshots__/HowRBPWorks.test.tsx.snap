// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test HowRBPWorks component should render the component 1`] = `
{
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "logTestingPlaygroundURL": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
}
`;

exports[`test HowRBPWorks component should render the list elements correctly 1`] = `
<div
  class="flex items-center"
  data-testid="list-element"
>
  <ul
    class="marker:mr-2 marker:pr-2 space-y-2 marker:leading-snug list-disc marker:text-blue-700 pb-8 pl-2 text-sm list-inside"
    start="-1"
  >
    <li>
      the age you would like to retire.
    </li>
    <li>
      your current income.
    </li>
    <li>
      how much you spend, including household bills and paying into your pension.
    </li>
    <li>
      an email address if you wish to save and come back later. 
    </li>
  </ul>
</div>
`;
